File: frontend/src/types/team.ts
/**
 * チーム関連の TypeScript 型定義。
 *
 * Member、Team、TeamSummary インターフェースを定義し、
 * Claude Code Agent Teams のチーム構成を型安全に管理します。
 *
*/

/**
 * チームメンバー情報を表すインターフェース。
 *
 * エージェントID、名前、タイプ、モデル、ステータス等の属性を持ちます。
 * Claude Code で作成されたエージェントの情報を表現します。
 */
export interface Member {
  agentId: string;
  name: string;
  agentType: string;
  model: string;
  joinedAt: number;
  tmuxPaneId?: string;
  cwd: string;
  subscriptions: string[];
  color?: string;
  status: 'active' | 'idle';
  lastActivity?: string;
}

/**
 * チーム詳細情報を表すインターフェース。
 *
 * チーム名、説明、作成日時、リードエージェント、メンバーリストを持ちます。
 * Claude Code で作成されたチームの完全な情報を表現します。
 */
export interface Team {
  name: string;
  description?: string;
  createdAt: number;
  leadAgentId: string;
  leadSessionId?: string;
  members: Member[];
  lastActivity?: string;
}

/**
 * チーム一覧表示用のサマリーインターフェース。
 *
 * チーム名、メンバー数、ステータスを持ち、一覧画面での高速表示用に
 * 最適化されています。
 *
 * モデル情報を含める場合は `models` プロパティを追加してください。
 */
export interface TeamSummary {
  name: string;
  description?: string;
  memberCount: number;
  status: 'active' | 'inactive';
  lastActivity?: string;
  leadAgentId: string;
  /** モデル使用状況（オプション） */
  models?: string[];
  /** プライマリモデル（最も使用されているモデル） */
  primaryModel?: string;
}

File: backend/app/models/team.py
"""チーム関連の Pydantic データモデル。

Member、Team、TeamSummary の3つのモデルを定義し、
Claude Code Agent Teams のチーム構成管理をサポートします。

"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class Member(BaseModel):
    """チームメンバーのデータモデル。

    エージェントID、名前、タイプ、モデル、参加日時、ステータス（active/idle）、
    色設定、tmux ペインID、作業ディレクトリ等を含みます。


    """

    agentId: str
    name: str
    agentType: str
    model: str = "unknown"
    joinedAt: int = 0
    tmuxPaneId: Optional[str] = ""
    cwd: str = ""
    subscriptions: list[str] = []
    color: Optional[str] = None
    status: str = "idle"  # active, idle
    lastActivity: Optional[datetime] = None
    prompt: Optional[str] = None  # エージェントのプロンプト
    backendType: Optional[str] = None  # バックエンドタイプ


class Team(BaseModel):
    """チームの完全なデータモデル。

    チーム名、説明、作成日時、リードエージェントID、セッションID、
    メンバーリストを含みます。Claude Code で作成されたチームの詳細情報です。


    """

    name: str
    description: Optional[str] = ""
    createdAt: int = 0
    leadAgentId: str
    leadSessionId: Optional[str] = ""
    members: list[Member] = []
    lastActivity: Optional[datetime] = None


class TeamSummary(BaseModel):
    """チーム一覧表示用のサマリーモデル。

    チーム名、説明、メンバー数、タスク数、ステータス、リードエージェントIDを含みます。
    一覧画面での高速表示に最適化された軽量モデルです。

    ステータス一覧:
    - active: セッションログ mtime ≤ 1時間
    - stopped: セッションログ mtime > 1時間
    - unknown: セッションログなし
    - inactive: members 空配列

    """

    name: str
    description: Optional[str] = ""
    memberCount: int
    taskCount: int = 0  # チームに関連するタスク数
    status: str  # active, inactive, stopped, unknown
    lastActivity: Optional[datetime] = None
    leadAgentId: str
    createdAt: Optional[int] = None  # 作成日時（Unixタイムスタンプ、ソート用）
